common/go-utils/src/inv.rs

```diff
@@ -1,12 +1,10 @@
-use crate::vrs::{Requirement, Version, VersionParseError};
+use crate::vrs::{Requirement, Version};
 use serde::{Deserialize, Serialize};
 use std::fs;
 use toml;
 
-const GITHUB_API_URL: &str = "https://api.github.com";
-const GO_REPO_NAME: &str = "golang/go";
+const GO_RELEASES_URL: &str = "https://go.dev/dl/?mode=json&include=all";
```

Comment:
Nice. Wish I had found this url. The GitHub logic was a bit more challenging.

Reply:
For sure, took some digging before I found the "include=all" parameter (in an old Go PR, but turns out [it's also documented here](https://pkg.go.dev/golang.org/x/website/internal/dl#section-documentation).

We could potentially also use/there might be a point in using the same endpoint when downloading the binaries (e.g. `https://go.dev/dl/go1.22.0.linux-amd64.tar.gz`), but it just redirects to the URL we currently use (`https://dl.google.com/go/go1.22.0.linux-amd64.tar.gz`) - and doesn't do any validation on file names/versions (e.g. `https://go.dev/dl/go123.456.789.linux-amd64.tar.gz` will still return an HTTP 304).

In any case, it's nice that we can get all the releases and checksums in one request (instead of fetching checksums for each file individually)... And not get hit by GitHub API limits when updating/diffing the inventory :)